src/doc/rustc/src/platform-support/fuchsia.md

```diff
@@ -0,0 +1,280 @@
+
+```
+hello_fuchsia.rs
+hello_fuchsia.cml
+package
+┣━ bin
+┃  ┗━ hello_fuchsia
+┣━ meta
+┃  ┣━ package
+┃  ┗━ hello_fuchsia.cm
+┗━ hello_fuchsia.manifest
+```
+
+Make the `package`, `package/bin`, and `package/meta` directories and create the
+following files inside:
+
+**`package/meta/package`**
+```json
+{"name":"hello_fuchsia","version":0}
+```
+
+The `package` file describes our package's name and version number. Every
+package must contain one.
+
+**`package/hello_fuchsia.manifest`**
+```
+bin/hello_fuchsia=${WORK_DIR}/bin/hello_fuchsia
+lib/ld.so.1=${SDK_PATH}/arch/${ARCH}/sysroot/dist/lib/ld.so.1
+lib/libfdio.so=${SDK_PATH}/arch/${ARCH}/lib/libfdio.so
+meta/package=${WORK_DIR}/meta/package
+meta/hello_fuchsia.cm=${WORK_DIR}/meta/hello_fuchsia.cm
+```
+
+Where `${WORK_DIR}` is the **absolute** path to the root of your directory.
+
+The `.manifest` file will be used to describe the contents of the package by
+relating their location when installed to their location on the file system. You
+can use this to make a package pull files from other places, but for this
+example we'll just be placing everything in the `package` directory.
+
+### Compiling a binary
+
+Using your freshly compiled `rustc`, you can compile a binary for Fuchsia using
+the following options:
+
+* `--target x86_64-fuchsia`/`--target aarch64-fuchsia`: Targets the Fuchsia
+  platform of your choice
+* `-L ${SDK_PATH}/arch/${ARCH}/lib`: Link against Fuchsia libraries from the SDK
+* `-L ${SDK_PATH}/arch/${ARCH}/sysroot/lib`: Link against Fuchsia kernel
+  libraries from the SDK
+
+Putting it all together:
+
+```sh
+# Configure these for the Fuchsia target of your choice
+# TARGET_ARCH={x86_64-fuchsia|aarch64-fuchsia}
+# ARCH={x64|aarch64}
+
+rustc --target ${TARGET_ARCH} -L${SDK_PATH}/arch/${ARCH}/lib -L${SDK_PATH}/arch/${ARCH}/sysroot/lib -o package/bin/hello_fuchsia hello_fuchsia.rs
+```
+
+### Bulding a component
+
+On Fuchsia, components require a component manifest written in CML. Here's a
+basic one that can run our single binary:
+
+**`hello_fuchsia.cml`**
+```
+{
+    include: [ "syslog/client.shard.cml" ],
+    program: {
+        runner: "elf",
+        binary: "bin/hello_fuchsia",
+    },
+}
+```
+
+Now we can compile that CML into a component manifest:
+
+```sh
+${SDK_PATH}/tools/${ARCH}/cmc compile hello_fuchsia.cml --includepath ${SDK_PATH}/pkg -o package/meta/hello_fuchsia.cm
+```
+
+`--includepath` tells the compiler where to look for `include`s from our CML.
+In our case, we're only using `syslog/client.shard.cml`.
+
+### Building a package
+
+Next, we'll build our package as defined by our manifest:
+
+```sh
+${SDK_PATH}/tools/${ARCH}/pm -o hello_fuchsia -m package/hello_fuchsia.manifest build
+```
+
+And archive it into a single file:
+
+```sh
+${SDK_PATH}/tools/${ARCH}/pm -o hello_fuchsia -m package/hello_fuchsia.manifest archive
+```
+
+This will produce `hello_fuchsia/hello_fuchsia-0.far` which is the package in
+the Fuchsia Archive Format (FAR).
```

Comment:
As an alternative to this step, you can produce an output manifest by passing `--output-package-manifest $out_pkg_manifest` to `pm build` and then publish that with `pm publish --repo repo -lp -f <(echo "$out_pkg_manifest")`

Package archives produced by `pm archive` are only needed if you need to like email the package to someone; if you're just planning to publish it into a package repository it's not necessary (and is slower, since it has to make an extra copy of all the files)